lib/xAxis.js

```diff
@@ -1,44 +1,17 @@
+import d3 from 'd3';
 
+export default function(xScale, configuration, where) {
+    const tickFormatData = configuration.tickFormat.map(t => t.slice(0));
+    const tickFormat = configuration.locale ? configuration.locale.timeFormat.multi(tickFormatData) : d3.time.format.multi(tickFormatData);
 
-module.exports = function (d3, config, xScale, graph, graphHeight, where) {
-  var xAxis = {};
-  var xAxisEls = {};
+    const axis = d3.svg.axis()
+        .scale(xScale)
+        .orient(where)
+        .tickFormat(tickFormat);
 
-  var tickFormatData = [];
+    if (typeof configuration.axisFormat === 'function') {
+        configuration.axisFormat(axis);
+    }
 
-  config.tickFormat.forEach(function (item) {
-    var tick = item.slice(0);
-    tickFormatData.push(tick);
-  });
-
-  var tickFormat = config.locale ? config.locale.timeFormat.multi(tickFormatData) : d3.time.format.multi(tickFormatData);
-  xAxis[where] = d3.svg.axis()
-    .scale(xScale)
-    .orient(where)
-    .tickFormat(tickFormat)
-  ;
-
-  if (typeof config.axisFormat === 'function') {
-    config.axisFormat(xAxis);
-  }
-
-  var y = (where == 'bottom' ? parseInt(graphHeight) : 0) + config.margin.top - 40;
-
-  xAxisEls[where] = graph
```

Comment:
No, it is now in the `Drawer`. An `xAxis` should not keep track of a collection of axes. 

By the way, the `Drawer` is far too heavy. It will be splitted in several smaller responsabilities classes.